src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/AES/AesCornerTests.cs

```diff
@@ -324,5 +324,115 @@ public static void ZeroLengthTransformPKCS7()
                 }
             }
         }
+
+#if netstandard17
+        [Fact]
+        public static void PaddingMode_Validation()
+        {
+            byte[] key = "1ed2f625c187b993256a8b3ccf9dcbfa5b44b4795c731012f70e4e64732efd5d".HexToByteArray();
+            byte[] iv = "47d1e060ba3c8643f9f8b65feeda4b30".HexToByteArray();
+
+            byte[] plainText = "".HexToByteArray();
+            ValidatePaddingMode(plainText, Array.Empty<byte>(), PaddingMode.Zeros, 0); // no block is added in this case!
+            ValidatePaddingMode(plainText, "d5450767bcc31793fe5065251b96b715".HexToByteArray(), PaddingMode.PKCS7, 16);
+            ValidatePaddingMode(plainText, "a3d32a3a9dca71b6f961f5a8ed7e414f".HexToByteArray(), PaddingMode.ANSIX923, 16);
+            ValidatePaddingMode(plainText, Array.Empty<byte>(), PaddingMode.ISO10126, 16);
+
+            plainText = "e505a2".HexToByteArray();
+            ValidatePaddingMode(plainText, "0a2e62938b03e5822ee251117a4ce066".HexToByteArray(), PaddingMode.Zeros, 13);
+            ValidatePaddingMode(plainText, "46785bde46622b92ff7c8ebb91508a4d".HexToByteArray(), PaddingMode.PKCS7, 13);
+            ValidatePaddingMode(plainText, "43b27d41a9fde73ca5db22c0fda76cb1".HexToByteArray(), PaddingMode.ANSIX923, 13);
+            ValidatePaddingMode(plainText, Array.Empty<byte>(), PaddingMode.ISO10126, 13);
+
+            plainText = "46785bde46622b92ff7c8ebb91508a".HexToByteArray();
+            ValidatePaddingMode(plainText, "1be8aa365a15d11fc7826b3a10602d09".HexToByteArray(), PaddingMode.Zeros, 1);
+            ValidatePaddingMode(plainText, "db5b7829cce732bfe609140cf45a8843".HexToByteArray(), PaddingMode.PKCS7, 1);
+            ValidatePaddingMode(plainText, "db5b7829cce732bfe609140cf45a8843".HexToByteArray(), PaddingMode.ANSIX923, 1);
```

Comment:
Can we get these as [Theory] on the ValidatePaddingMode method?

```c#
[Theory]
[InlineData(PaddingMode.Zeros, 1, "46785bde46622b92ff7c8ebb91508a", "1be8aa365a15d11fc7826b3a10602d09")]
[InlineData(PaddingMode.PKCS7, 1, "46785bde46622b92ff7c8ebb91508a", "db5b7829cce732bfe609140cf45a8843")]
[InlineData(PaddingMode.ANSIX923, 1, "46785bde46622b92ff7c8ebb91508a", null)]
```

(Yeah, I switched the argument order to make the test case name start off with the relevant parts)